allennlp/models/ensemble/bidaf_ensemble.py

```diff
@@ -0,0 +1,155 @@
+            ``span_end_logits`` to find the most probable span.  Shape is ``(batch_size, 2)``
+            and each offset is a token index.
+        best_span_str : List[str]
+            If sufficient metadata was provided for the instances in the batch, we also return the
+            string from the original passage that the model thinks is the best answer to the
+            question.
+        """
+
+        subresults = [submodel(question, passage, span_start, span_end, metadata) for submodel in self.submodels]
+
+        batch_size = len(subresults[0]["best_span"])
+
+        output = {
+                "best_span": torch.zeros(batch_size, 2).long(),
+                "best_span_str": []
+        }
+        for index in range(batch_size):
+            best_index = ensemble(index, subresults)
+            best_span = subresults[best_index]["best_span"].data[index].long()
+            output["best_span"][index] = best_span
+
+            if metadata is not None:
+                best_span_str = subresults[best_index]["best_span_str"][index]
+                output["best_span_str"].append(best_span_str)
+
+                answer_texts = metadata[index].get('answer_texts', [])
+                if answer_texts:
+                    self._squad_metrics(best_span_str, answer_texts)
+
+        return output
+
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+        exact_match, f1_score = self._squad_metrics.get_metric(reset)
+        return {
+                'em': exact_match,
+                'f1': f1_score,
+        }
+
+    @classmethod
+    def from_params(cls, vocab: Vocabulary, params: Params):
+        if vocab:
+            raise ConfigurationError("vocab should be None")
+
+        submodels = []
+        paths = params.pop("submodels")
+        for path in paths:
+            submodels.append(load_archive(path).model)
+
+        return cls(submodels=submodels)
+
+def ensemble(index: int, subresults: List[Dict[str, torch.Tensor]]) -> int:
+    """
+    Identifies the best prediction given the results from the submodels.
+
+    Parameters
+    ----------
+    index : int
+        The index within this index to ensemble
+
+    subresults : List[Dict[str, torch.Tensor]]
+
+    Returns
+    -------
+    The index of the best submodel.
+    """
+
+    # Populate span_votes so each key represents a span range that a submodel predicts and the value
+    # is the number of models that made the prediction.
+    spans = [(subresult["best_span"].data[index][0], subresult["best_span"].data[index][1])
+             for subresult in subresults]
+    votes: Dict[Tuple[int, int], int] = {span:spans.count(span) for span in spans}
+
+    # Choose the majority-vote span.
+    # If there is a tie, break it with the average confidence (span_start_probs + span_end_probs).
+    options = []
+    for i, subresult in enumerate(subresults):
+        start = subresult["best_span"].data[index][0]
```

Comment:
You _should_ be able to just do something like:

```python
span_start_probs = torch.sum(subresult['span_start_probs'] for subresult in subresults) / len(subresults)
span_end_probs = torch.sum(subresult['span_end_probs'] for subresult in subresults) / len(subresults)
best_span = BidirectionalAttentionFlow._get_best_span(span_start_probs.log(), span_end_probs.log())
```
Doing it this way is differentiable, which allows training (either the submodels themselves, or adding some weights to the sum above that get learned).